### PR TITLE
feat: add option to paste plain text

### LINKED ISSUE #7023: [FR] Copy Pasting Raw Mark Down text
### Impact

- Everyone will be benefiting from this change, for instance the application itself encourages using markdown text, by when typing markdown automatically recognizing and converting into stylized text, we should extend going in the direction of markdown text usage.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by asjqkkkk**
To fix #7023 

### Feature Preview

- add `Paste as plain text` item to ContextMenu
- pasting the original Markdown data will automatically convert it

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart [hunk 1]
```diff
@@ -467,6 +469,52 @@ void main() {
...
+  });
+
+  testWidgets('paste markdowns as plain', (tester) async {
+    const markdown = '''
+# I'm h1
+## I'm h2
+### I'm h3
+#### I'm h4
+##### I'm h5
+###### I'm h6''';
```
Comment (Xazin) [inline frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart:503]: Can move this `markdown` const outside of this scope to make it available to the `paste markdowns` test above.

File: frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart [hunk 2]
```diff
@@ -467,6 +469,52 @@ void main() {
...
+      plainText: markdown,
+      pasteAsPlain: true,
+      (editorState) {
+        final children = editorState.document.root.children;
+        expect(children.length, 6);
+        for (var i = 0; i < children.length; i++) {
+          final text = children[i].delta!.toPlainText();
+          final expectText = '${'#' * (i + 1)} I\'m h${i + 1}';
+          expect(text, expectText);
+        }
```
Comment (Xazin) [inline frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart:514]:
```suggestion
        for (int i = 1; i <= children.length; i++) {
          final text = children[i].delta!.toPlainText();
          final expectText = '${'#' * i} I\'m h$i';
          expect(text, expectText);
        }
```

File: frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart [hunk 3]
```diff
@@ -467,6 +469,52 @@ void main() {
...
+      (editorState) {
+        final children = editorState.document.root.children;
+        expect(children.length, 6);
+        for (var i = 0; i < children.length; i++) {
+          final text = children[i].delta!.toPlainText();
+          expect(
+            text,
+            'I\'m h${i + 1}',
+          );
+        }
```
Comment (Xazin) [inline frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart:491]:
```suggestion
        for (int i = 1; i <= children.length; i++) {
          final text = children[i].delta!.toPlainText();
          expect(text, 'I\'m h$i');
        }
```

File: frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart (hunk 1 above)
Comment (asjqkkkk) [reply frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart]: All are good suggestions.

**Review by Xazin — APPROVED** [review]

File: frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart [hunk 4]
```diff
@@ -27,10 +26,13 @@ void main() {
       await tester.pasteContent(
         plainText: List.generate(lines, (index) => 'line $index').join('\n'),
         (editorState) {
-          expect(editorState.document.root.children.length, 3);
+          expect(editorState.document.root.children.length, 1);
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart:29]: there should be 3 lines here.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/copy_and_paste/custom_paste_command.dart [hunk 5]
```diff
@@ -190,3 +214,22 @@ Future<bool> _pasteAsLinkPreview(
...
+
+  // dispatch the paste event
+  final data = await getIt<ClipboardService>().getData();
+  final plainText = data.plainText;
+  if (plainText != null && plainText.isNotEmpty) {
+    await editorState.pastePlainText(plainText);
+    return Log.info('Pasted plain text');
+  }
+
+  return Log.info('unable to parse the clipboard content');
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/copy_and_paste/custom_paste_command.dart:234]:
```suggestion
  // dispatch the paste event
  final data = await getIt<ClipboardService>().getData();
  final plainText = data.plainText;
  if (plainText != null && plainText.isNotEmpty) {
    await editorState.pastePlainText(plainText);
    Log.info('Pasted plain text');
    return;
  }

  Log.info('unable to parse the clipboard content');
  return;
```

File: frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart (hunk 4 above)
Comment (asjqkkkk) [reply frontend/appflowy_flutter/integration_test/desktop/document/document_copy_and_paste_test.dart:29]: Yes, that's right

**Pull request merged**